test/ng/directive/ngOptionsSpec.js

```diff
@@ -922,6 +922,25 @@ describe('ngOptions', function() {
       expect(scope.selected).toEqual([20]);
       expect(element).toEqualSelectValue([20], true);
     });
+
+
+    it('should ignore array values beginning with $$', function() {
+      scope.arr = ['one', 'two', '$$hashKey'];
```

Comment:
Performance consideration aside, I don't think this solves the actual issue anyway.
It's indeed the `keys` we need to look for `$$` not the values.